src/test/java/se/iths/jakartaeelab1/dto/PersonDtoTest.java

```diff
@@ -0,0 +1,76 @@
+package se.iths.jakartaeelab1.dto;
+
+import jakarta.validation.*;
+import org.hibernate.validator.messageinterpolation.ParameterMessageInterpolator;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import se.iths.jakartaeelab1.entity.Person;
+
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+
+
+public class PersonDtoTest {
+    private static Validator validator;
+
+    @BeforeAll
+    static void setupValidator() {
+        ValidatorFactory factory = Validation.byDefaultProvider()
+                .configure()
+                .messageInterpolator(new ParameterMessageInterpolator())
+                .buildValidatorFactory();
+        validator = factory.getValidator();
+    }
+
+    @Test
+    void map_shouldMapPersonToPersonDto() {
+        Person person = new Person();
+        person.setName("John");
+        person.setAge(25);
+        person.setProfession("Developer");
+
+        PersonDto personDto = PersonDto.map(person);
+
+        assertEquals("John", personDto.name());
+        assertEquals(25, personDto.age());
+        assertEquals("Developer", personDto.profession());
+
+        PersonDto mappedPersonDto = PersonDto.map(person);
+
+        Set<ConstraintViolation<PersonDto>> violations = validator.validate(mappedPersonDto);
+
+        assertTrue(violations.isEmpty(), "Validation should pass for mapping Person to PersonDto");
+    }
+
+    @Test
+    void map_shouldMapPersonDtoToPerson() {
+        PersonDto personDto = new PersonDto(
+                "Jane",
+                30,
+                "Engineer");
+
+        Person person = PersonDto.map(personDto);
+
+        assertEquals("Jane", person.getName());
+        assertEquals(30, person.getAge());
+        assertEquals("Engineer", person.getProfession());
+
+        PersonDto personDtoInput = new PersonDto("Jane", 30, "Engineer");
+        Set<ConstraintViolation<PersonDto>> violations = validator.validate(personDtoInput);
+
+        assertTrue(violations.isEmpty(), "Validation should pass for mapping PersonDto to Person");
+
+        Person mappedPerson = PersonDto.map(personDtoInput);
+
+        assertEquals("Jane", mappedPerson.getName());
+        assertEquals(30, mappedPerson.getAge());
+        assertEquals("Engineer", mappedPerson.getProfession());
+    }
```

Comment:
It looks like you added validation to the tests for the map-methods. Is it not better to split these up, so that we have four tests? One for each map-method. One for testing a valid PersonDto and one for testing an invalid PersonDto. As Martin said, one test should test one thing.

Reply:
if I try to split the tests up i get errors on two of them:

@Test
    void mapPersonDtoToPerson_invalidInput_shouldFailValidation() {
        PersonDto personDto = new PersonDto(null, 30, "Engineer");

        assertThrows(ConstraintViolationException.class, () -> {
            PersonDto.map(personDto);
        });
    }

@Test
    void mapPersonToPersonDto_invalidInput_shouldFailValidation() {
        Person person = new Person();

        assertThrows(ValidationException.class, () -> {
            PersonDto.map(person);
        });
    }

I get this eror message:
org.opentest4j.AssertionFailedError: Expected jakarta.validation.ValidationException to be thrown, but nothing was thrown.

Reply:
🤔 this will surely help [1.3. Validating constraints](https://docs.jboss.org/hibernate/stable/validator/reference/en-US/html_single/#_validating_constraints) and also [Lecture 2024-02-27 timestamp 01:18:30](https://ithogskolan.sharepoint.com/:v:/s/Java23/EYQl8KRMyW9EpRfDPnPZoWEBeql6j-Uulvxccsaw7fvXBg?e=sprHX2&nav=eyJyZWZlcnJhbEluZm8iOnsicmVmZXJyYWxBcHAiOiJTdHJlYW1XZWJBcHAiLCJyZWZlcnJhbFZpZXciOiJTaGFyZURpYWxvZy1MaW5rIiwicmVmZXJyYWxBcHBQbGF0Zm9ybSI6IldlYiIsInJlZmVycmFsTW9kZSI6InZpZXcifSwicGxheWJhY2tPcHRpb25zIjp7InN0YXJ0VGltZUluU2Vjb25kcyI6NDcxMC4yNH19)